_posts/2021-04-22-release-1.13.0.md

```diff
@@ -0,0 +1,450 @@
+
+---
+
+## Table API/SQL
+
+### Table-valued functions for Windows
+
+Defining time windows is one of the most frequent operations in streaming SQL queries. We updated
+the Flink SQL window syntax and semantics to be
+closer to the SQL standard. This is in sync with other vendors in the industry by expressing windows
+as a kind of table-valued functions.
+
+TUMBLE and HOP windows have been ported to the new syntax. SESSION windows will follow in a
+subsequent release. A new CUMULATE window function can assigns windows with an expanding step size
+until the maximum window size is reached:
+
+```sql
+SELECT window_time, window_start, window_end, SUM(price) AS total_price 
+  FROM TABLE(CUMULATE(TABLE Bid, DESCRIPTOR(bidtime), INTERVAL '2' MINUTES, INTERVAL '10' MINUTES))
+GROUP BY window_start, window_end, window_time;
+```
+
+By referencing window start and window end of the table-valued window functions, it is now possible
+to compute Top N windowed aggregations next to regular windowed aggregations and windowed joins:
+
+```sql
+SELECT window_time, ...
+  FROM (
+    SELECT *, ROW_NUMBER() OVER (PARTITION BY window_start, window_end ORDER BY total_price DESC) 
+      as rank 
+    FROM t
+  ) WHERE rank <= 100;
+```
+
+### Improved interoperability between DataStream and Table API/SQL 
+
+The Table API becomes equally important to the DataStream API which is why a good interoperability
+between the two APIs is crucial. (FLIP-136): The core Row class has received a major update with new
+toString/hash/equals methods. Fields can not only be accessed by position but also by name with
+support of sparse representations. The new methods `StreamTableEnvironment.toDataStream/fromDataStream`
+can model the DataStream API as a regular table source or sink with a smooth type system and event-time
+integration.
+
+```java
+Table table=tableEnv.fromDataStream(
+	dataStream,Schema.newBuilder()
+	.columnByMetadata("rowtime","TIMESTAMP(3)")
+	.watermark("rowtime","SOURCE_WATERMARK()")
+	.build());
+
+DataStream<Row> dataStream=tableEnv.toDataStream(table)
+	.keyBy(r->r.getField("user"))
+	.window(...)
+```
+
+### Improved SQL Client 
+
+During the past releases, the SQL Client's feature set could not keep up with the evolution of the
+TableEnvironment in the Table API. In this release, the SQL Client has been updated for feature
+parity with the programmatic APIs.
+
+*Easier Configuration and Code Sharing*
+
+The support of YAML files to configure the SQL Client will be discontinued. Instead, the client
+accepts one or more initialization scripts to configure an empty session. Additionally, a user can
+declare a main SQL script for executing multiple statements in one file.
+
+```
+./sql-client.sh -i init1.sql init2.sql -f sqljob.sql
+```
+
+*More Commands and Support for Statement Sets*
+
+For multi-statement execution, commands such as ADD JAR, STATEMENT SET, and improved SET/RESET
```

Comment:
Sorry, `ADD JAR` is not supported in this release, it is postponed to next release, also including `REMOVE JAR`, `SHOW JARS`.